ddtrace/tracer.py

```diff
@@ -147,20 +154,37 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
             parent = None
 
         if parent:
-            # this is a child span
+            trace_id = parent.trace_id
+            parent_span_id = parent.span_id
+            sampling_priority = parent.get_sampling_priority()
+        else:
+            trace_id, parent_span_id, sampling_priority = context.get_context_attributes()
```

Comment:
This doesn't work with the current API. We don't have a way to `set_call_context()` (the complete API was proposed to OpenTracing but we never made that change for ourselves), so it means that we can't:
```python
ctx = Context(trace_id=trace_id, parent_id=parent_id, sampling_priority= sampling_priority)
tracer.set_call_context(ctx)  # we don't have this API
```

This means that `context.get_context_attributes()` returns always the value of the first created `Context` when you `start_span()` or `trace()`. Because the `Context` is mutable, to get the right values users should:
```python
ctx = tracer.get_call_context()
ctx.set_context_attributes(...)
# use start_span() or trace() as usual
```

Of course the fact that `Context` is mutable, is something that is planned to be changed.

Reply:
This line wasn't asserting that trace_id / parent_id / sampling_priority could change over time nor that we would need a set_call_context.
This `context. get_context_attributes` was from the case we create a root span from a context (from a remote tid/sid/priority, like in the iohttp change) instead of a parent.